storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1370,9 +1469,974 @@ public static int toPositive(int number) {
+    }
+
+    public static int execCommand(String command) throws ExecuteException, IOException {
+        String[] cmdlist = command.split(" ");
+        CommandLine cmd = new CommandLine(cmdlist[0]);
+        for (int i = 1; i < cmdlist.length; i++) {
+            cmd.addArgument(cmdlist[i]);
+        }
+
+        DefaultExecutor exec = new DefaultExecutor();
+        return exec.execute(cmd);
+    }
+
+    /**
+     * Extra dir from the jar to destdir
+     *
+     * @param jarpath
+     * @param dir
+     * @param destdir
+     *
+    (with-open [jarpath (ZipFile. jarpath)]
+    (let [entries (enumeration-seq (.entries jarpath))]
+    (doseq [file (filter (fn [entry](and (not (.isDirectory entry)) (.startsWith (.getName entry) dir))) entries)]
+    (.mkdirs (.getParentFile (File. destdir (.getName file))))
+    (with-open [out (FileOutputStream. (File. destdir (.getName file)))]
+    (io/copy (.getInputStream jarpath file) out)))))
+
+     */
+    public static void extractDirFromJar(String jarpath, String dir, String destdir) {
+        JarFile jarFile = null;
+        FileOutputStream out = null;
+        InputStream in = null;
+        try {
+            jarFile = new JarFile(jarpath);
+            Enumeration<JarEntry> jarEnums = jarFile.entries();
+            while (jarEnums.hasMoreElements()) {
+                JarEntry entry = jarEnums.nextElement();
+                if (!entry.isDirectory() && entry.getName().startsWith(dir)) {
+                    File aFile = new File(destdir, entry.getName());
+                    aFile.getParentFile().mkdirs();
+                    out = new FileOutputStream(aFile);
+                    in = jarFile.getInputStream(entry);
+                    IOUtils.copy(in, out);
+                    out.close();
+                    in.close();
+                }
+            }
+        } catch (IOException e) {
+            LOG.info("Could not extract {} from {}", dir, jarpath);
+        } finally {
+            if (jarFile != null) {
+                try {
+                    jarFile.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(
+                            "Something really strange happened when trying to close the jar file" + jarpath);
+                }
+            }
+            if (out != null) {
+                try {
+                    out.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(
+                            "Something really strange happened when trying to close the output for jar file" + jarpath);
+                }
+            }
+            if (in != null) {
+                try {
+                    in.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(
+                            "Something really strange happened when trying to close the input for jar file" + jarpath);
+                }
+            }
+        }
```

Comment:
The try/catch everything can be made a lot simpler with the java7 AutoClosable and make it match the clojure code better.

```
    public static void extractDirFromJar(String jarpath, String dir, String destdir) {
        try (JarFile jarFile = new JarFile(jarpath);) {
            Enumeration<JarEntry> jarEnums = jarFile.entries();
            while (jarEnums.hasMoreElements()) {
                JarEntry entry = jarEnums.nextElement();
                if (!entry.isDirectory() && entry.getName().startsWith(dir)) {
                    File aFile = new File(destdir, entry.getName());
                    aFile.getParentFile().mkdirs();
                    try(
                        out = new FileOutputStream(aFile);
                        in = jarFile.getInputStream(entry); ) {
                        IOUtils.copy(in, out);
                    }
                }
            }
        } catch (IOException e) {
            LOG.info("Could not extract {} from {}", dir, jarpath);
        }
    }
```